lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,256 @@
+        percentiles looked up correspond to the probabilities of these
+        thresholds being reached.
+
+        Snow-fall level::
+
+            Reference field: Percentiled snow fall level (m ASL)
+            Other field: Orography (m ASL)
+
+            300m ----------------- 30th Percentile snow fall level
+            200m ----_------------ 20th Percentile snow fall level
+            100m ---/-\----------- 10th Percentile snow fall level
+            000m --/---\----------  0th Percentile snow fall level
+            ______/     \_________ Orogaphy
+
+        The orography heights are compared against the heights that
+        correspond with percentile values to find the band in which they
+        fall; this diagram hides the 2-dimensional variability of the snow
+        fall level. The percentile values are then interpolated to the
+        height of the point being considered. This constructs a
+        2-dimensional field of probabilities that snow will be falling at
+        each point in the orography field.
+    """
+
+    def __init__(self, percentiles_cube, output_name=None):
+        """
+        Initialise class. Sets an inverse_ordering (bool) switch to true for
+        cases where the percentiled data increases in the opposite sense to the
+        percentile coordinate.
+                e.g.  0th Percentile - Value = 10
+                     10th Percentile - Value = 5
+                     20th Percenitle - Value = 0
+
+        Args:
+            percentiles_cube (iris.cube.Cube):
+                The percentiled field from which probabilities will be obtained
+                using the input cube. This cube should contain a percentiles
+                dimension, with fields of values that correspond to these
+                percentiles. The cube passed to the process method will contain
+                values of the same diagnostic (e.g. height) as this reference
+                cube.
+            output_name (str):
+                The name of the cube being created,
+                e.g.'probability_of_snowfall'.
+        """
+        self.percentile_coordinate = find_percentile_coordinate(
+            percentiles_cube)
+        self.percentiles_cube = percentiles_cube
+
+        if output_name is not None:
+            self.output_name = output_name
+        else:
+            self.output_name = "probability_of_{}".format(
+                percentiles_cube.name())
+
+        # Set inverse_ordering switch
+        percentile_slices = percentiles_cube.slices_over(
+            self.percentile_coordinate)
+        self.inverse_ordering = False
+        first_percentile = percentile_slices.next().data
+        for percentile_values in percentile_slices:
+            last_percentile = percentile_values.data
+        if (first_percentile - last_percentile >= 0).all():
+            self.inverse_ordering = True
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
```

Comment:
I'm not sure whether you need to print out the `inverse_ordering` in the __repr__ or not. It might be useful.

Reply:
✅